### PR TITLE
Google auth

### PR DESCRIPTION
**Pull request by hecerinc**
- Adds Google Auth 
- Finish setting up the users table migration
- Adds user to table, validates Google token, authenticates or registers new user.

## Instructions to run this PR

1. Go to https://curl.haxx.se/docs/caextract.html
2. Download the `cacert.pem`
3. Save it somewhere in your computer (where you won't move it) (Recommendation: `C:/xampp`)
4. Add this line to your `php.ini` configuration file:

```bash
curl.cainfo = "C:\xampp\php\extras\ssl\cacert.pem" # Your path goes here
```

5. Run `composer install` in your directory
6. Have fun

### CODE REVIEW

File: resources/views/layouts/app.blade.php [hunk 1]
```diff
@@ -28,6 +28,7 @@
 			</div>
 		</div>
 		<div class="container">
+			<?php var_dump(Auth::user()) ?>
```
Comment (osdagoso) [inline resources/views/layouts/app.blade.php]: Why is this here?

Comment (hecerinc) [reply resources/views/layouts/app.blade.php]: Pa que sepan cómo verificar que haya un usuario.
